browser/profiles/brave_profile_manager_browsertest.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <vector>
+
+#include "base/strings/utf_string_conversions.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/chrome_notification_types.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_attributes_entry.h"
+#include "chrome/browser/profiles/profile_attributes_storage.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "content/public/test/test_utils.h"
+
+using BraveProfileManagerTest = InProcessBrowserTest;
+
+namespace {
+
+// An observer that returns back to test code after a new profile is
+// initialized.
+void OnUnblockOnProfileCreation(base::RunLoop* run_loop,
+                                Profile* profile,
+                                Profile::CreateStatus status) {
+  if (status == Profile::CREATE_STATUS_INITIALIZED)
+    run_loop->Quit();
+}
+
+struct TestProfileData {
+  base::string16 profile_name;
+  base::string16 profile_name_expected_after_migration;
+  bool force_default_name;
+  base::FilePath profile_path;
+};
+
+std::vector<TestProfileData> GetTestProfileData(
+    ProfileManager* profile_manager) {
+  const std::vector<TestProfileData> profile_data = {
+    {
+      base::ASCIIToUTF16("Person 1"),
+      base::ASCIIToUTF16("Profile 1"), true,
+      profile_manager->user_data_dir().Append(
+          profile_manager->GetInitialProfileDir())},
+    {
+      base::ASCIIToUTF16("Person 2"),
+      base::ASCIIToUTF16("Profile 2"), true,
+      profile_manager->user_data_dir().Append(
+          FILE_PATH_LITERAL("testprofile2"))},
+    {
+      base::ASCIIToUTF16("ZZCustom 3"),
+      base::ASCIIToUTF16("ZZCustom 3"), false,
+      profile_manager->user_data_dir().Append(
+          FILE_PATH_LITERAL("testprofile3"))},
+  };
+  return profile_data;
+}
+
+}  // namespace
+
+// Test that legacy profile names (Person X) that have
+// not been user-modified are automatically renamed
+// to brave profile names (Profile X).
+IN_PROC_BROWSER_TEST_F(BraveProfileManagerTest, PRE_Migrateprofile_names) {
```

Comment:
we should avoid underscore as best practice https://github.com/google/googletest/blob/master/googletest/docs/faq.md#why-should-test-suite-names-and-test-names-not-contain-underscore unless we are using special prefix like `DISABLED`, `MANUAL`, `MAYBE`, `PRE`...etc. so this should be like `PRE_MigrateProfileNames`

Reply:
@darkdh yeah as stated above, that was due to some over-eager find/replace! That's fixed now via https://github.com/brave/brave-core/compare/a0d154f196ae541be59dd286f6d21b11006b5318..af4f1e901aba92a3e19ac4263a95368c777b7fe9